documentation/debugging-staticanalysis-errors.md

```diff
@@ -0,0 +1,47 @@
+- [Where to find StaticAnalysis reports](#where-to-find-staticsanalysis-reports)
+- [Types of StaticAnalysis Errors](#types-of-staticanalysis-errors)
+    - [Breaking Changes](#breaking-changes)
+    - [Signature Issues](#signature-issues)
+    - [Help Issues](#help-issues)
+
+## How to know if you have a StaticAnalysis Error
+If your build is failing, click on the Jenkins job inside the PR (marked as "Default" within checks).  Then check the Console Output within the Jenkins job.  If you have this error, then you have failed StaticAnalysis:
+```
+d:\workspace\powershell\build.proj(511,5): error MSB3073: The command "d:\workspace\powershell\src\Package\StaticAnalysis.exe d:\workspace\powershell\src\Package\Debug d:\workspace\powershell\src\Package true false" exited with code 255.
+```
+
+## Where to find StaticAnalysis reports
+
+The StaticAnalysis reports could show up in two different places in the CI build:
+- On the status page in Jenkins, under the Build Artifacts: the relevant files are BreakingChangeIssues.csv, SignatureIssues.csv, and/or HelpIssues.csv.
+- On the status page in Jenkins, click Build Artifacts then navigate to src/Package.  You will see BreakingChangeIssues.csv, SignatureIssues.csv, and/or HelpIssues.csv.
+
+Locally, the StaticAnalysis report will show up under Azure-PowerShell/src/Package. You will see BreakingChangeIssues.csv, SignatureIssues.csv, and/or HelpIssues.csv.  You can generate these files by running
+```
+msbuild build.proj
+```
+
+## Types of StaticAnalysis Errors
+The three most common Static Analysis errors are breaking changes, signature issues, and help issues.  To figure out which type of error is causing the build failure, open each of the relevant .csv files (if the .csv file does not exist, there is no violation detected).  Any issue marked with severity 0 or 1 must be resolved in order for the build to pass.
+
+### Breaking Changes
+If you make a change that could cause a breaking change, it will be listed in BreakingChangeIssues.csv.  Please look at each of these errors, and if they do indeed introduce a breaking change in a non-breaking change release, please revert the change that caused this violation.  Sometimes the error listed in the .csv file can be a false positive (for example, if you change a parameter attribute to span all parameter sets rather than individual parameter sets).  Please read the error thoroughly and examine the relevant code before deciding that an error is a false positive, and contact the Azure PowerShell team if you have questions.  If you are releasing a preview module, are releasing during a breaking change release, or have determined that the error is a false positive, please follow these instructions:
+- Navigate to the [BreakingChangeIssues.csv file](https://github.com/Azure/azure-powershell/blob/preview/tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv) in our repo.
```

Comment:
@maddieclayton I'd reorder these steps (for both _Breaking Changes_ and _Signature Issues_) so that the user knows to copy the contents from the **build** `.csv` file first and then open the **repository** `.csv` file.

I would also call out that you need to open these `.csv` files and edit them in a text editor (such as VS Code). Opening them and copying/pasting the lines use Excel will cause parsing errors during StaticAnalysis.

Reply:
Done